xc/device_common.ml

```diff
@@ -238,6 +238,14 @@ let readdir ~xs d = try xs.Xs.directory d with Xs_protocol.Enoent _ -> []
 let to_list ys = List.concat (List.map Opt.to_list ys)
 let list_kinds ~xs dir = to_list (List.map parse_kind (readdir ~xs dir))
 
+let filter_on_devids ~xs ?for_devids dir =
+  match for_devids with
+  | None -> List.filter_map parse_int (readdir ~xs dir)
+  | Some devids ->
+    devids |>
+    List.filter (fun devid->
+        try ignore(xs.Xs.read (sprintf "%s/%d" dir devid)); true with _->false)
```

Comment:
There is no need to pipe here, why not
```ocaml
List.filter (fun devid->
    try ignore(xs.Xs.read (sprintf "%s/%d" dir devid)); true
    with _ -> false
) devids
```
instead?

Reply:
And preserve the comment.

Reply:
Indeed